addons/event/models/mail_template.py

```diff
@@ -20,3 +20,9 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
         if self.env.context.get('filter_template_on_event'):
             args = expression.AND([[('model', '=', 'event.registration')], args])
         return super(MailTemplate, self)._name_search(name, args, operator, limit, name_get_uid)
+
+    def unlink(self):
+        domain = ('template_ref', 'in', [f"{template._name},{template.id}" for template in self])
+        for model in ['event.mail', 'event.type.mail']:
+            self.env[model].sudo().search([domain]).unlink()
```

Comment:
Would do this after the call to super, unlink could fail in which case this search/unlink will just be useless work

Domain could probably be inlined too.